### DISCUSSION TITLE
Trace context propagation is not happening with tomcat and http-url-connection instrumentation disabled

### DISCUSSION THREAD

**Discussion by srinusanduri** [open-telemetry/opentelemetry-java-instrumentation#7852] [Q&A] [answered]
This documentation https://opentelemetry.io/docs/instrumentation/java/automatic/agent-config/ mentions that we can disable all libraries auto instrumentation by default using property `otel.instrumentation.common.default-enabled=false` and then enable specific instrumentations of our interest using property `otel.instrumentation.[name].enabled=true`. However, context propagation is not happening if boundary library instrumentations are disabled. 

When I disabled tomcat instrumentation using property `otel.instrumentation.tomcat.enabled=false`, parent span context coming from upstream service is lost and following library in the flow within the service (spring-webmvc in my case) not creating span though its instrumentation is enabled (`otel.instrumentation.spring-web.enabled=true`).
Similarly if I disable `http-url-connection` instrumentation using property `otel.instrumentation.http-url-connection.enabled=false` trace context is not propagated to downstream service.

Looks like tracing is not working without boundary library instrumentation enabled. If this the right expectation, the documentation may have to be updated accordingly.

In my case, I want spring-webmvc span to be root span. To achieve this, I tried disabling tomcat instrumentation so that spring-webmvc can start the root span but this is not working. Is it possible to achieve this? Thank you for your time.

**Comment by laurit** [comment, marked as answer]
It is not possible. Spring web instrumentation does not do context propagation nor does it create a server span.
Disabling all instrumentations and enabling only the ones you need is an advanced configuration option. You'll have to understand what these instrumentations do and which you really need.
